### PR TITLE
[#1958] Use syntax coloring for code blocks in docs

### LINKED ISSUE #1958: Use syntax coloring for code blocks in docs
### What problem(s) did you run into that caused you to request additional documentation? What, if any, existing documentation relates to this proposal?

Even though [MarkBind offers syntax coloring for code blocks](https://markbind.org/userGuide/formattingContents.html#code), it is not currently being used in some places in the docs, such as the style guide & learning the basics sections of the DG. This makes the code quite difficult to read.

### Example

**Left = current (no syntax coloring), right = with syntax coloring**

ch-1
ch-2

### What would you like added or modified in the documentation?

We can add syntax coloring by simply appending the programming language after the 3 backticks at the start of the code block. 
e.g. \``` -> \```java
This will improve readability of the docs.

### PR DESCRIPTION
**Pull request by asdfghjkxd**
Fixes #1958

### Proposed commit message

```
Current code blocks do not use the built-in syntax highlighting
feature offered by Markbind/Markdown.

This impedes the readability of the code within the code blocks.

Let's update the code blocks in the documentation to enable
syntax highlighting to further enhance readability of the documentation
and the code.
```

### Other information

N/A

### CODE REVIEW

File: docs/dg/learningBasics.md [hunk 1]
```diff
@@ -358,15 +358,15 @@ Here are some small tasks for you to gain some basic knowledge of the code relat
...
+  ```css
   .active-text {
     color: mui-color('green');
   }
   ```
 
   In `c-summary-charts.vue`, locate `summary-chart__title--name`, and add the following to its `v-bind:class` attribute map.
 
-  ```
+  ```vue
```
Comment (MarcusTXK) [inline docs/dg/learningBasics.md:369]: I don't believe MarkBind supports the `vue` code block, perhaps we could either use another code block syntax highlighting, or remove it altogether.

**Review by jonasongg — APPROVED** [review]
LGTM!

**Review by MarcusTXK — CHANGES_REQUESTED** [review]
Nice job locating where the code blocks are used in the documentation. Just some minor nitpicks to address before this is merged.

File: docs/dg/styleGuides.md [hunk 2]
```diff
@@ -48,16 +48,16 @@ In addition to what has been mentioned in the [**Java** coding standard (SE-EDU)
   * This is not necessary (although still recommended) for methods with `@Override` annotations if Javadoc is used. However, if the method that is being overriden is part of your code and has Javadoc, all parameters must be described.
 
 Negative Examples:
-```
-Not okay (Only mentions zoneId parameter):
+```java
+//Not okay (Only mentions zoneId parameter):
```
[the remark this answers is not in the document]
Comment (asdfghjkxd) [reply docs/dg/styleGuides.md]: Oh oops, I must have forgotten to add the space after changing all the code blocks. It should be fixed in the new commit!

File: docs/dg/learningBasics.md (hunk 1 above)
Comment (asdfghjkxd) [reply docs/dg/learningBasics.md]: I see, I have removed the `vue` tag from the code, it should look better now!

**Review by MarcusTXK — APPROVED** [review]
LGTM

**Review by vvidday — APPROVED** [review]
LGTM!

Looks like Markbind [uses highlight.js](https://github.com/MarkBind/markbind/blob/master/packages/core/package.json#L45) for syntax highlighting, and it [doesn't seem to support Vue. ](https://highlightjs.readthedocs.io/en/latest/supported-languages.html)

Also since we are using Pug + Vue, it probably won't be straightforward to configure syntax highlighting for our Vue files. I think we can just leave those blocks as is.

**Pull request merged**